modules/adwords_appengine/src/main/java/com/google/api/ads/adwords/jaxws/utils/v201802/SelectorBuilderImpl.java

```diff
@@ -0,0 +1,472 @@
+    Collections.addAll(this.fields, fields);
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl fields(EntityField... fields) {
+    return fields(this.enumFieldsToNames(fields));
+  }
+
+  /**
+   * Returns the array of Strings resulting from calling {@code name()} on each field.
+   *
+   * @param fields the array of fields as Java Enums.
+   */
+  private String[] enumFieldsToNames(EntityField... fields) {
+    String[] fieldsNames = new String[fields.length];
+    for (int i = 0; i < fields.length; i++) {
+      fieldsNames[i] = fields[i].name();
+    }
+    return fieldsNames;
+  }
+
+  @Override
+  public SelectorBuilderImpl increaseOffsetBy(int additionalOffset) {
+    if (this.paging == null) {
+      this.paging = new Paging();
+    }
+    Integer startIndex = this.paging.getStartIndex();
+    if (startIndex == null) {
+      startIndex = 0;
+    }
+    this.paging.setStartIndex(startIndex + additionalOffset);
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl limit(int limit) {
+    if (this.paging == null) {
+      this.paging = new Paging();
+    }
+    this.paging.setNumberResults(limit);
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl offset(int offset) {
+    if (this.paging == null) {
+      this.paging = new Paging();
+    }
+    this.paging.setStartIndex(offset);
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl removeLimitAndOffset() {
+    this.paging = null;
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl orderAscBy(String field) {
+    OrderByWrapper order = new OrderByWrapper();
+    order.setField(field);
+    order.setSortOrder(SortOrder.ASCENDING);
+
+    this.ordering.add(order);
+
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl orderAscBy(EntityField field) {
+    return this.orderAscBy(field.name());
+  }
+
+  @Override
+  public SelectorBuilderImpl orderDescBy(String field) {
+    OrderByWrapper order = new OrderByWrapper();
+    order.setField(field);
+    order.setSortOrder(SortOrder.DESCENDING);
+
+    this.ordering.add(order);
+
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl orderDescBy(EntityField field) {
+    return this.orderDescBy(field.name());
+  }
+
+  @Override
+  public SelectorBuilderImpl removeOrderBy(String field) {
+    Iterator<OrderByWrapper> iterator = this.ordering.iterator();
+    while (iterator.hasNext()) {
+      OrderByWrapper order = iterator.next();
+      if (order.getField().equals(field)) {
+        iterator.remove();
+      }
+    }
+    return this;
+  }
+
+  @Override
+  public SelectorBuilderImpl removeOrderBy(EntityField field) {
+    return this.removeOrderBy(field.name());
+  }
+
+  @Override
+  public SelectorBuilderImpl forDateRange(DateTime start, DateTime end) {
+    if (this.dateRange == null) {
+      this.dateRange = new DateRange();
+    }
+    DateFormat dateFormat = new SimpleDateFormat(DEFAULT_DATE_FORMAT);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.